src/adagrams.js

```diff
@@ -1,15 +1,214 @@
+const LETTERPOOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+
+const createLetterList = () => {
+  const letters = [];
+  const letterList = [];
+  for (const [key, value] of Object.entries(LETTERPOOL)) {
+    letters.push(key.repeat(value)) ;
+  }
+  for (const ele of letters) {
+    for (const letter of ele) {
+      letterList.push(letter);
+    }
+  }
+  
+  return letterList;
+};
+
+
+
+const pool = createLetterList();
+
+const createRandomizer = () => {
+  const randomIndex = Math.floor(Math.random() * (pool.length + 1));
+  const randomLetter = pool[randomIndex];
+  return randomLetter
+};
+
+
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  // const pool = createLetterList(LETTERPOOL);
+  const drawnLetters = [];
+
+  while (drawnLetters.length < 10) {
+    const randomLetter = createRandomizer();
+    const occurences = drawnLetters.filter( letter => letter === randomLetter).length
```

Comment:
Love how you used the `filter` method here, it is one of the most utilized Javascript methods, especially in React as you probably already seen.